src/inference/Inference.jl

```diff
@@ -667,21 +667,24 @@ julia> sum(abs2, ys_test - ys_pred) ≤ 0.1
 true
 ```
 """
-function predict(model::Turing.Model, chain::MCMCChains.Chains; include_all = false)
+function predict(model::Model, chain::MCMCChains.Chains; include_all = false)
     spl = DynamicPPL.SampleFromPrior()
 
     # Sample transitions using `spl` conditioned on values in `chain`
-    transitions = transitions_from_chain(model, chain; sampler = spl)
-
-    # Let the Turing internals handle everything else for you
-    chain_result = AbstractMCMC.bundle_samples(
-        Distributions.GLOBAL_RNG,
-        model,
-        spl,
-        length(chain),
-        transitions,
-        MCMCChains.Chains
-    )
+    chain_result = mapreduce(MCMCChains.chainscat, MCMCChains.chains(chain)) do chain_idx
+        transitions = transitions_from_chain(
+            model, chain[:, :, chain_idx];
+            sampler = spl
+        )
+        AbstractMCMC.bundle_samples(
+            Distributions.GLOBAL_RNG,
```

Comment:
We should use
```suggestion
            Random.GLOBAL_RNG,
```
Even better would be to add a `rng` argument as first argument to `predict` which would be set to `Random.GLOBAL_RNG` if omitted.

Reply:
Btw, is the `rng` actually used for anything in MCMCChains?

Reply:
Actually, one thing we've forgot is that`transitions_from_chain` ought to have `rng`, right? Do you pass it to `evaluate` for the `Model`?

Reply:
Every time we run the model we use an RNG (if not specified, it's the default one). The current approach is to omit it when (we assume that) all samples are provided. However, maybe we should always use one explicitly since users could also make use of in Turing-external code (when they call an SDE solver e.g.)

Reply:
> The current approach is to omit it when (we assume that) all samples are provided.

Not sure what you mean here :confused: I mean to ask if `model = ...; model(rng, ...)` was the way to go (which it seems like it is). One thing though, it doesn't seem like calling `model(rng)` works but `model(rng, varinfo)` does. Is that an issue or am I missing something?

Reply:
> However, maybe we should always use one explicitly since users could also make use of in Turing-external code (when they call an SDE solver e.g.)

Inside the model you mean? If so, I agree!

Reply:
I mean in multiple implementions (e.g., when computing the elementwise loglikelihoods) we do not pass an explicit RNG since we assume that there is no randomness if all variables are provided.

> Is that an issue or am I missing something?

Yes, the dispatches in https://github.com/TuringLang/DynamicPPL.jl/blob/161f8200c5d608c481b58e056c2687a380f1c0d5/src/model.jl#L79-L92 should be changed (the issue can be fixed by just defining `(model::Model)(rng, varinfo = VarInfo(), ...)` instead, i.e., adding the default value to the method definition).

Reply:
> Inside the model you mean? If so, I agree!

Yep, inside the model.